# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1387867049 18000
#      Tue Dec 24 01:37:29 2013 -0500
# Branch stable
# Node ID 525af54479ebf126e5a7561874ebde910d813c4e
# Parent  65e4e0569ed4ed2aa368f0048561f0b9c19d1f81
allow run-octave to find octave-gui program

* run-octave.in: Set OCTAVE_ARCHLIB when executing Octave so that
octave-gui will be found.

diff --git a/run-octave.in b/run-octave.in
--- a/run-octave.in
+++ b/run-octave.in
@@ -73,17 +73,23 @@ if [ $# -gt 0 ]; then
     driver="strace -o octave.trace"
     shift
   elif [ "x$1" = "x-cli" ]; then
     octave_executable="$builddir/src/octave-cli"
     shift
   fi
 fi
 
+## We set OCTAVE_ARCHLIBDIR so that the wrapper program can find the
+## octave-gui program in the build tree.  That will fail if we ever
+## need Octave to find other things in ARCHLIBDIR that are not built
+## in the $builddir/src directory.
+
 OCTAVE_BINDIR="$builddir/src" \
+OCTAVE_ARCHLIBDIR="$builddir/src" \
 OCTAVE_SITE_INITFILE="$top_srcdir/scripts/startup/main-rcfile" \
 OCTAVE_DEFAULT_QT_SETTINGS="$builddir/libgui/default-qt-settings" \
 OCTAVE_LOCALE_DIR="$builddir/libgui/languages" \
 OCTAVE_JAVA_DIR="$builddir/scripts/java" \
   exec $builddir/libtool --mode=execute $driver \
     "$octave_executable" --no-init-path --path="$LOADPATH" \
     --image-path="$IMAGEPATH" --doc-cache-file="$DOCFILE" \
     --built-in-docstrings-file="$BUILT_IN_DOCSTRINGS_FILE" \
